### ISSUE TITLE
Add support for Flutter applications

### ISSUE THREAD

**Issue by pedromassango** [shamblett/coap#4]
This package can't be used with Flutter because one of their dependency uses Dart mirrors wich is not supported by Flutter applications.

This is what I receive while trying to use this packagein a Flutter application:

```
E/flutter ( 9165): [ERROR:flutter/shell/common/shell.cc(199)] Dart Error: error: import of dart:mirrors is not supported in the current Dart runtime
E/flutter ( 9165): [ERROR:flutter/shell/common/engine.cc(198)] Could not prepare to run the isolate.
E/flutter ( 9165): [ERROR:flutter/shell/common/engine.cc(137)] Engine not prepare and launch isolate.
E/flutter ( 9165): [ERROR:flutter/shell/common/shell.cc(437)] Could not launch engine with configuration.
```

Edit:
This can solved by removing `safe_config` from the dependencies since it uses Dart mirrors and use tool like code-generator to generate the configuration class from a `.yaml` file.

**Comment by pedromassango** [comment]
This is related to https://github.com/shamblett/coap/issues/3

**Comment by shamblett** [comment]
Yes OK but the package doesn't say it can be used with Flutter, its marked as Dart Native on pub, it's not a simple as removing safe_config as its the base class for the configuration options. I'll see if I can find another.

**Comment by smaziano** [comment]
Hello @shamblett,
Thanks for sharing your work. We found your implementation interesting. You're OK with a pull request to support it ?

**Comment by shamblett** [comment]
Yes no problem.

**Comment by shamblett** [comment]
Package re published at 2.0.0
